sql/parser/type_check.go

```diff
@@ -228,6 +231,63 @@ func typeCheckComparisonOp(op ComparisonOp, dummyLeft, dummyRight Datum) (Datum,
 		dummyLeft.Type(), op, dummyRight.Type())
 }
 
+func typeCheckConditionalExpr(expr *ConditionalExpr) (Datum, error) {
+	switch expr.Operator {
+	case If:
+		if len(expr.Exprs) != 3 {
```

Comment:
Not sure what you mean by statically. The grammar certainly never creates an `IF` expressions containing a different number of expressions. Would an `IfExpr` that implements `ConditionalExpr` only have slots for 3 expressions? Then we would need different code in `WalkExpr`.